core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1178,6 +1178,33 @@ class LogManager(logDirs: Seq[File],
     }
   }
 
+  def recoverAbandonedFutureLogs(brokerId: Int, newTopicsImage: TopicsImage): Unit = {
+    val abandonedFutureLogs = findAbandonedFutureLogs(brokerId, newTopicsImage)
+    abandonedFutureLogs.foreach { case (futureLog, currentLog) =>
+      val tp = futureLog.topicPartition
+      if (cleaner != null) {
```

Comment:
> I guess cleaner.abortAndPauseCleaning is added because we call resumeCleaning later, and it will cause error if we don't call abortAndPauseCleaning here?

That's correct if the cleaning operation hasn't started. The cleaning operation is scheduled on a separate thread so we cannot be sure if `inProgress` map in `LogCleanerManager` has a key for the given topicPartition at the time we iterate over these logs.

Reply:
Maybe we need to add comments for `abortAndPauseCleaning`. The `recoverAbandonedFutureLogs` is called in starting and calling `abortAndPauseCleaning` here is a bit weird to me

Reply:
Added a comment in [3d78e55](https://github.com/apache/kafka/pull/15136/commits/3d78e5567fd16aa77f0b9f8c510ad26f0e6443c6)